platform/mellanox/docker-syncd-mlnx/lib/port_utils.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+
+from python_sdk_api.sx_api import *
+import inspect
+
+DEVICE_ID = 1
+SWITCH_ID = 0
+ETHERNET_PREFIX = 'Ethernet'
+ASIC_MAX_LANES = {SX_CHIP_TYPE_SPECTRUM: 4, SX_CHIP_TYPE_SPECTRUM2: 4, 
+                  SX_CHIP_TYPE_SPECTRUM3: 8, SX_CHIP_TYPE_SPECTRUM4: 8}
+
+def sx_get_ports_map(handle):
+    """ Get ports map from SDK logical index to SONiC index
+    
+    Args:
+        handle (sx_api_handle_t): SDK handle
+    
+    Returns:
+        dict : Dictionary of ports indices. Key is SDK logical index, value is SONiC index (4 for Ethernet4)
+    """         
+    try:
+        ports_map = {}
+        
+        # Get chip type
+        chip_type = sx_get_chip_type(handle)
+        
+        # Get ports count
+        port_cnt_p = new_uint32_t_p()
+        rc = sx_api_port_device_get(handle, DEVICE_ID, SWITCH_ID, None,  port_cnt_p)
+        sx_check_rc(rc)
+        
+        # Get ports
+        port_cnt = uint32_t_p_value(port_cnt_p)
+        port_attributes_list = new_sx_port_attributes_t_arr(port_cnt)
+        rc = sx_api_port_device_get(handle, DEVICE_ID, SWITCH_ID, port_attributes_list,  port_cnt_p)
+        sx_check_rc(rc)
+        
+        for i in range(0, port_cnt):
+            port_attributes = sx_port_attributes_t_arr_getitem(port_attributes_list, i)
+            label_port = port_attributes.port_mapping.module_port
+            logical_port = port_attributes.log_port;
+            lane_bmap = port_attributes.port_mapping.lane_bmap;
+            
+            if (is_phy_port(port_attributes.log_port) == False):
+                continue
+            
+            # Calculate sonic index (sonic index=4 for Ethernet4)
+            lane_index = get_lane_index(lane_bmap, ASIC_MAX_LANES[chip_type])
+            assert lane_index != -1, "Failed to calculate port index"
+            
+            sonic_index = label_port * ASIC_MAX_LANES[chip_type] + lane_index;
```

Comment:
Check if label port start from 1 or 0

Reply:
Label port is the port number printed on the front panel minus one.
On ACS-MSN3700C with 32 ports, first label port is 0 and last is 31 (total of 32 as expected).